### ISSUE TITLE
Streamline prop names

### ISSUE THREAD

**Issue by CarloPalinckx** [MyOnlineStore/bricks#290]
### Is your feature request related to a problem? Please describe ⚡️
Currently a lot of props do not really align in their naming conventions. To name a few:

- `action`/`closeAction` vs `onClick`/`onClose`
- `show` vs `isOpen`

### Describe the solution you'd like ✨
My preferred prop conventions would be:
- `onClick` because this is a common convention and it makes it clear that you are supposed to react on that "event" being triggered. In case of `onClose` this will not actually close a modal for instance, but you trigger a state change with it that closes it.
- `show`/`open` etc. since there is a bit of divide in the community on prop like this. I suggest we go for the most terse option.

**Comment by woutsluiter** [comment]
Good point. I would also go for `onClick`/`onClose` and `show`/`open`. I take it this would also applies to the state / internal variables? Some other candidates to follow this convention:
```
hasError > error // Though with this name I'd expect an error object, not a boolean
isLoading > loading
isRound > round
isFree > free
isSelected > selected
isTargeted > targeted
isFirst > first
isLast > last
```

Would be good to describe these conventions in the Contribute docs.

**Comment by woutsluiter** [comment]
And I'd prefer `show` over `open`. Not sure why.

**Comment by CarloPalinckx** [comment]
I would keep internals as is. They just seems a bit verbose in the context of JSX imo.

**Comment by CarloPalinckx** [comment]
Resolved with #316
